src/backend/tcop/postgres.c

```diff
@@ -988,6 +990,12 @@ exec_simple_query(const char *query_string)
 			break;
 		}
 
+		if (IsA(parsetree, SelectStmt) && IsAdhocQuery(parsetree))
```

Comment:
Obviously this can interfere quite a bit with the standard execution path. Perhaps the best way to initialize an adhoc CQ would be with a function call:

    SELECT exec_adhoc_continuous_query('SELECT ... FROM ...');

This would avoid having to change any syntax and would make adhoc detection reliable, as everything would be isolated within the body of a single function.